x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -363,17 +369,6 @@ public void authenticate(final AuthenticationToken authenticationToken, final Ac
                 listener.onResponse(AuthenticationResult.unsuccessful(msg, null));
                 return;
             }
-            final List<String> groups = this.claimParserGroups.getClaimValues(claimsSet);
-            final Map<String, Object> userMetadata;
-            try {
-                userMetadata = this.populateUserMetadata ? JwtUtil.toUserMetadata(jwt) : Map.of();
-            } catch (Exception e) {
-                final String msg = "Realm [" + super.name() + "] parse metadata failed for principal=[" + principal + "].";
-                final AuthenticationResult<User> unsuccessful = AuthenticationResult.unsuccessful(msg, e);
-                LOGGER.debug(msg, e);
-                listener.onResponse(unsuccessful);
-                return;
-            }
```

Comment:
I don't think that is correct. I had the same question early on. OIDC does not include JWT claims for delegated authz, and JWT realm mirrors that implementation.

Here is the relevant code from OIDC realm. When delegated authz is on, OIDC realm uses a listener to get `auth.getMetadata()` coming back from the delegated authz realm. OIDC realm only adds `_oidc_tokendata`, JWT claims are not included.

https://github.com/elastic/elasticsearch/blob/1c65f1c2855392f43aa5bae891cd71033c706b6b/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectRealm.java#L196-L211

Reply:
So, OIDC only returns JWT claims during role mapping, not delegated authz. JWT realm is doing the same.

If you think OIDC and JWT realms should return JWT claims in both use cases, I think we can raise it in a separate issue for discussion. I think it would be a good idea too, but it is not in scope for this PR.

For this PR, scope is only to add User attributes for mail, name, and DN. User metadata handling is unchanged.